### PR TITLE
Cumulus 2489: Update Deployment & Configuration Docs

### PR DESCRIPTION
**Pull request by jjmccoy**
**Summary:** Summary of changes

Addresses [CUMULUS-2489: Clear guided documentation for Deployment & Configuration](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2489)

## Changes

* Update 'Getting Started', 'FAQs', 'Glossary', 'Deployment', and 'Integrator Guide' sections for Wiki & GitHub Docs - Terraform focused
* Reference with links to Terraform modules and variable examples
* Change 'Deployment' sidebar to guide a user better

## PR Checklist

- [x] Update CHANGELOG
- [ ] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: docs/faqs.md [hunk 1]
```diff
@@ -63,6 +63,24 @@ Below are some commonly asked questions that you may encounter that can assist y
...
+<details>
+  <summary>Is Cumulus customizable and flexible?</summary>
+
+  Answer: Yes. Cumulus is a modular architecture that allows you to decide which components that you want/need to deploy. These components are maintained as Terraform modules.
+</details>
+
+<details>
+  <summary>What are Terraform modules?</summary>
+
+  Answer: They are modules that are composed to create a Cumulus deployment, which gives integrators the flexibility to choose the components of Cumulus that want/need. To view Cumulus maintained modules or steps on how to create a module go to [Terraform modules](../tf-modules).
```
Comment (laurenfrederick) [inline docs/faqs.md:75]: We don't have a `tf-modules` page. Which page are you trying to link to?

**Review by laurenfrederick — CHANGES_REQUESTED** [review]
Just one comment

File: docs/faqs.md (hunk 1 above)
Comment (jjmccoy) [reply docs/faqs.md:75]: @laurenfrederick Doh! I put in the placeholder and forgot to reference this link: https://github.com/nasa/cumulus/tree/master/tf-modules. I will add that.

Comment (jjmccoy) [reply docs/faqs.md]: @laurenfrederick When I went back to work on it, I realized that this resource is external to the docs. Updated link.

**Review by laurenfrederick — APPROVED** [review]

File: docs/faqs.md (hunk 1 above)
Comment (laurenfrederick) [reply docs/faqs.md]: Thank you!

**Pull request merged**